src/plugins/validation/2and3/semantic-validators/schema-ibm.js

```diff
@@ -511,3 +524,44 @@ function isRootSchema(path) {
     current === 'schema' || (parent === 'definitions' && path.length === 2)
   );
 }
+
+// Flag any properties that have the same name but an inconsistent type.
+function checkProperties(
+  schema,
+  propertiesToCompare,
+  contextPath,
+  configOption,
+  messages
+) {
+  if (!schema.properties) return;
+
+  for (const [key, value] of Object.entries(schema.properties)) {
+    // skips properties that use $ref
+    if (value.type) {
+      if (Object.prototype.hasOwnProperty.call(propertiesToCompare, key)) {
```

Comment:
This isn't necessary in this scenario. You simplify it like this:

```suggestion
      if (propertiesToCompare.key) {
```